Mention the `rustup target add` step in the initial cross-compile assistance message

If `cargo libcnb package` (or libcnb-test, once #704 is fixed) detects that the required cross-compilation toolchain is not installed, it prints the help messages here:
https://github.com/heroku/libcnb.rs/blob/a9759eef8e356f6396f1df8b16afe591645251d5/libcnb-package/src/cross_compile.rs#L9-L91

That help message explains how to install the musl-tools package (or Homebrew equivalent), after which the packaging steps will get further, but will then fail again with errors of form:

```
test expected_pack_failure ... ok
   Compiling component-a v0.0.0 (/Users/emorley/src/libcnb.rs/libcnb-test/tests/fixtures/buildpacks/component-a)
test build_workspace_component_buildpack ... FAILEDomponent-a(bin)                                                                                       
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-linux-musl` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-unknown-linux-musl`

error: cannot find macro `println` in this scope
 --> src/main.rs:2:5
  |
2 |     println!("Buildpack A");
  |     ^^^^^^^

error: requires `sized` lang_item

For more information about this error, try `rustc --explain E0463`.
```

Whilst this next set of errors is better than nothing (it at least suggests the next command that needs running, `rustup target add x86_64-unknown-linux-musl`):
- the command the user needs to run is somewhat buried inside the rest of the error message
- it has meant twice in a row the user tried to run a command and it has failed - giving the impression of a never-ending set of undocumented setup steps

For a more pleasant UX, we should suggest running the `rustup target add x86_64-unknown-linux-musl` command in the original compile assistance message. Worst case the user reinstalls an already added target (which will just be a no-op), or misses the second install suggestion, and proceeds to see the error above anyway.

GUS-W-14438920.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Improved the consistency of all user-facing libcnb.rs error message wordings. ([#722](https://github.com/heroku/libcnb.rs/pull/722))
+- The assistance error message shown when the necessary cross-compilation tools are not found now also includes the `rustup target add` step. ([#729](https://github.com/heroku/libcnb.rs/pull/729))
 
 ### Fixed
 

--- a/libcnb-package/src/cross_compile.rs
+++ b/libcnb-package/src/cross_compile.rs
@@ -25,7 +25,10 @@ The easiest way to install the required cross-compilation toolchain is to run:
 brew install messense/macos-cross-toolchains/x86_64-unknown-linux-musl
 
 For more information, see:
-https://github.com/messense/homebrew-macos-cross-toolchains",
+https://github.com/messense/homebrew-macos-cross-toolchains
+
+You will also need to install the Rust target, using:
+rustup target add x86_64-unknown-linux-musl",
             )), |gcc_binary_path| {
                 CrossCompileAssistance::Configuration {
                     cargo_env: vec![
@@ -57,7 +60,10 @@ linker for the target platform must be installed on your computer.
 
 The easiest way to install 'musl-gcc' is to install the 'musl-tools' package:
 - https://packages.ubuntu.com/focal/musl-tools
-- https://packages.debian.org/bullseye/musl-tools",
+- https://packages.debian.org/bullseye/musl-tools
+
+You will also need to install the Rust target, using:
+rustup target add x86_64-unknown-linux-musl",
             )),
         }
     } else if target_triple.as_ref() == AARCH64_UNKNOWN_LINUX_MUSL && cfg!(target_os = "linux") {
@@ -82,7 +88,10 @@ linker for the target platform must installed on your computer.
 The easiest way to install the 'g++-aarch64-linux-gnu', 'libc6-dev-arm64-cross', and 'musl-tools' packages:
 - https://packages.ubuntu.com/focal/g++-aarch64-linux-gnu
 - https://packages.ubuntu.com/focal/musl-tools
-- https://packages.ubuntu.com/focal/libc6-dev-arm64-cross",
+- https://packages.ubuntu.com/focal/libc6-dev-arm64-cross
+
+You will also need to install the Rust target, using:
+rustup target add aarch64-unknown-linux-musl",
             )),
         }
     } else {

